src/OpenTelemetry/Metrics/MetricPoint.cs

```diff
@@ -247,12 +247,18 @@ internal void Update(double number)
                 case AggregationType.Histogram:
                     {
                         int i;
-                        for (i = 0; i < this.histogramBuckets.ExplicitBounds.Length; i++)
+                        if (double.IsNaN(number))
                         {
-                            // Upper bound is inclusive
-                            if (number <= this.histogramBuckets.ExplicitBounds[i])
+                            i = this.histogramBuckets.ExplicitBounds.Length;
+                        }
+                        else
+                        {
+                            // If number is not a histogram bound, then the result will be zero
```

Comment:
I think you meant "will be negative" right?

Reply:
Yes, I did!